include/librealsense2/dds/dds-device-broadcaster.h

```diff
@@ -0,0 +1,111 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <unordered_map>
+#include <vector>
+#include <atomic>
+#include <mutex>
+#include <condition_variable>
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/domain/DomainParticipantListener.hpp>
+#include <fastdds/dds/publisher/DataWriterListener.hpp>
+#include <librealsense2/rs.hpp>  // Include RealSense Cross Platform API
+#include <librealsense2/utilities/concurrency/concurrency.h>
+
+namespace librealsense {
+namespace dds {
+namespace topics {
+namespace raw {
+class device_info;
+}  // namespace raw
+class device_info;
+}  // namespace topics
+
+
+class dds_participant;
+
+class dds_device_broadcaster
+{
+public:
+    dds_device_broadcaster( librealsense::dds::dds_participant & participant );
+    ~dds_device_broadcaster();
+
+    // Start listening for device changes
+    bool run();
+
+    // Create a new data writer for a new connected device
+    void add_device( rs2::device dev );
+
+    // Destroy the removed device data writer
+    void remove_device( rs2::device dev );
+
+
+private:
+    // We want to know when readers join our topic
+    class dds_client_listener : public eprosima::fastdds::dds::DataWriterListener
```

Comment:
This can pro'ly be moved to the .cpp -- in next PR :)

Reply:
👍